### PR TITLE
Image in mediabag needs to be written in binary mode

### LINKED ISSUE #3982: rendering a document with online image to pdf is not possible on Windows
I tried to render the quarto example document "hello.qmd" to pdf. This throws an error, because an external image is linked in Markdown (the same problem would appear in R Markdown). As soon as I comment out the image the pdf document can be created.

I think it is not ideal to use an example document that cannot be rendered in all widely used formats html, pdf, docx.

### PR DESCRIPTION
**Pull request by cderv**
This is required on Windows OS and fixes #3982

We are also adding a `writeText()` function to keep version of write to file for text mode only (`mode = "a")`. 

I added a test but also took the opportunity to add `fileExists()` support for smoke-all. This was a way to test supporting file is correctly downloaded in mediabag, as we currently expect that. 

Hopefully that is ok, and not too much of a test. No rendering issue may be enough as LaTeX was the one throwing this issue

### CODE REVIEW

File: src/resources/pandoc/datadir/init.lua [hunk 1]
```diff
@@ -1789,7 +1791,12 @@ _quarto = {
    projectOffset = projectOffset,
    file = {
       read = read_file,
-      write = write_file,
+      write = function(path, contents) 
+         return write_file(path, contents, "ab")
```
Comment (cderv) [inline src/resources/pandoc/datadir/init.lua:1795]: Should it be instead ? 
```suggestion
         return write_file(path, contents, "wb") 
```

**Review by dragonstyle — COMMENTED** [review]
looks good other than that one naming tweak

File: src/resources/pandoc/datadir/init.lua [hunk 2]
```diff
@@ -1789,7 +1791,12 @@ _quarto = {
    projectOffset = projectOffset,
    file = {
       read = read_file,
-      write = write_file,
+      write = function(path, contents) 
+         return write_file(path, contents, "wb") 
+      end,
+      writeText = function(path, contents)
```
[the remark this answers is not in the document]
Comment (cderv) [reply src/resources/pandoc/datadir/init.lua:1797]: Yes sure. I don't know why I remember the otherway around !

**Comment by cderv** [comment]
Thanks @dragonstyle ! I went with merging myself as it was a small change;

**Comment by dragonstyle** [comment]
TY for diagnosing and fixing this!

**Pull request merged**